### ISSUE TITLE
iPad video subtitles pop-over is really small

### ISSUE THREAD

**Issue by BPerlakiH** [kiwix/kiwix-apple#915]
The subtitles pop-over is really small (compared to the native one, which is disabled by video js).

**Comment by BPerlakiH** [comment]
So far the only way I have found is to adjust videojs settings in the content itself, which requires some find a replace magic on the loaded HTML content, which might or might not work...

I think we can leave this issue as is, the user can switch between subtitle languages more easily in full screen mode, only the inline one is really small.

See here:

<img width="468" alt="Screenshot 2024-08-09 at 11 03 55" src="https://github.com/user-attachments/assets/13dfeed5-a1dd-4624-be39-f3dad1b50526">

compared to the full screen one:
![Screenshot 2024-08-09 at 11 00 28](https://github.com/user-attachments/assets/54f66946-e44c-45ff-870d-77688ac8be18)

**Comment by kelson42** [comment]
@BPerlakiH You mean you woukd like to have the popup box to have a bigger maximum height?

**Comment by BPerlakiH** [comment]
@kelson42 here is the comparison between using the provided videojs ui on the iPad,
and how it looks when we turn videojs off, and the native/system ui is used:

# videojs:

https://github.com/user-attachments/assets/418c5ef2-4429-42e9-a59c-7dd17cd869c6

----------------------------
# native:

https://github.com/user-attachments/assets/7082f84a-72ae-475c-8236-af7275d220fa

**Comment by BPerlakiH** [comment]
So to sum up, this is specific only to iPad, and only when the video is not in full screen, it is really hard to browse those languages, and you need to really accurate, as it's less than half of a finger size.

If the video is in full screen the subtitles can be changed without any problems.
Probably it's a small issue, and might be fixed in next video js releases.
(the new video ZIM provided in: #914 unfortunately has no subtitles, so couldn't use that for this purpose).

**Comment by kelson42** [comment]
@benoit74 Your feedback would be welcome here about (1) "issue" assessment (2) how the solution could/should look like.

**Comment by benoit74** [comment]
If I get the issue correctly, then I think that this should be reported to videoJS since a generic solution would help all users.

I don't have an iPad to test, but do we have the same issue on https://videojs.com/advanced?video=bipbop-advanced?

**Comment by benoit74** [comment]
And don't we have the same issue on all small touch screens? (e.g. on my Android phone, it is not easy to be sure to spot proper language either)

**Comment by kelson42** [comment]
@benoit74 Probably! To me this is an enhancement to do at videojs level and if they can't using a custom CSS to apply in our scraper(s). Therefore I would propose to:
* Move to `openzim/youtube`
* Open an issue upstream

Does that sounds appropriate?

**Comment by benoit74** [comment]
Why youtube ? video.js is used in many scrapers, not only youtube.

**Comment by kelson42** [comment]
> Why youtube ? video.js is used in many scrapers, not only youtube.

Once fixed in youtube scraper I will request to fix in other scrapers. One issue at a time.

**Comment by benoit74** [comment]
But this issue speaks about a problem in TED ...

**Comment by kelson42** [comment]
@benoit74 Oui, so move to `openzim/ted`. Does that change anything at the end (i believe all our scrapers are impacted in the manner, or isn't it the case)?

**Comment by benoit74** [comment]
All scrapers relying on video.js are probably impacted in the same manner indeed

**Comment by benoit74** [comment]
I have opened https://github.com/videojs/video.js/discussions/8892 to start the discussion at video.js side.

That been said, I don't get why once in fullscreen you have a kind of different player. I don't have the same behavior on Android in my browser or in the app. Is this something specific to kiwix-apple?

**Comment by kelson42** [comment]
@benoit74 @BPerlakiH It seems upstream (video-js) does not care, so we should probably fix it ourself. I would:
* Provide a PR to video-js PR
* If not moving forward at video-js, make a patch ourself in our ZIM files

**Comment by benoit74** [comment]
Yes, this is the approach I would like as well.

**Comment by kelson42** [comment]
Moving to `openzim/ted`

**Comment by kelson42** [comment]
Kamino closed and cloned this issue to [openzim/ted](https://github.com/openzim/ted/issues/236)
